generator/testdata/google/example/library/v1/library_service_stub.gapic.cc.baseline

```diff
@@ -146,199 +147,145 @@ class RetryLibraryServiceStub : public LibraryServiceStub {
   google::gax::Status
   CreateBook(google::gax::CallContext& context,
              ::google::example::library::v1::CreateBookRequest const& request,
-             ::google::example::library::v1::Book& response) override {
-    auto retry_policy = clone_retry(context);
-    auto backoff_policy = clone_backoff(context);
-    gax::MethodInfo info = context.Info();
-    while (true) {
-      // The next layer stub may add metadata, so create a
-      // fresh call context each time through the loop.
-      gax::CallContext context_copy(context);
-      gax::Status status = next_stub_->CreateBook(context_copy, request, response);
-      if (status.IsOk() || !retry_policy->OnFailure(status) ||
-          !info.idempotency == gax::MethodInfo::Idempotency::IDEMPOTENT) {
-        return status;
-      }
-
-      auto delay = backoff_policy->OnCompletion();
-      std::this_thread::sleep_for(delay);
-    }
+             ::google::example::library::v1::Book* response) override {
+    return MakeRetryCall<::google::example::library::v1::CreateBookRequest,
+                         ::google::example::library::v1::Book>(
+        context, request, response,
+        [=this](CallContext& c,
+                ::google::example::library::v1::CreateBookRequest const& req,
+                ::google::example::library::v1::Book* resp)
+            ->gax::Status {
```

Comment:
Do you need to define the return type for this lambda?  The default should work, right?

Reply:
I don't think it's necessary. The compiler should be able to deduce it.